modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/detail/CountriesDetail.js

```diff
@@ -0,0 +1,161 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import className from 'classnames';
+import PropTypes from 'prop-types';
+import React, {useContext, useEffect, useMemo, useRef} from 'react';
+
+import {
+	ChartDispatchContext,
+	ChartStateContext,
+	useDateTitle,
+	useIsPreviousPeriodButtonDisabled,
+} from '../../context/ChartStateContext';
+import ConnectionContext from '../../context/ConnectionContext';
+import {
+	StoreDispatchContext,
+	StoreStateContext,
+} from '../../context/StoreContext';
+import {generateDateFormatters as dateFormat} from '../../utils/dateFormat';
+import TimeSpanSelector from '../TimeSpanSelector';
+import TotalCount from '../TotalCount';
+import Countries from './Countries';
+
+export default function CountriesDetail({
+	currentPage,
+	handleDetailPeriodChange,
+	timeSpanOptions,
+	trafficShareDataProvider,
+	trafficSourcesDataProvider,
+	trafficVolumeDataProvider,
+}) {
+
+	const {languageTag} = useContext(StoreStateContext);
+
+	const dateFormatters = useMemo(() => dateFormat(languageTag), [
+		languageTag,
+	]);
+
+	const {firstDate, lastDate} = useDateTitle();
+
+	const title = useMemo(() => {
+		return dateFormatters.formatChartTitle([firstDate, lastDate]);
+	}, [dateFormatters, firstDate, lastDate]);
+
+	const dispatch = useContext(StoreDispatchContext);
+
+	const chartDispatch = useContext(ChartDispatchContext);
+
+	const {pieChartLoading, timeSpanKey, timeSpanOffset} = useContext(
+		ChartStateContext
+	);
+
+	const {validAnalyticsConnection} = useContext(ConnectionContext);
+
+	const isPreviousPeriodButtonDisabled = useIsPreviousPeriodButtonDisabled();
+
+	const firstUpdateRef = useRef(true);
+
+	const trafficSourceDetailClasses = className(
+		'c-p-3 traffic-source-detail',
+		{
+			'traffic-source-detail--loading': pieChartLoading,
+		}
+	);
+
+	useEffect(() => {
+		if (firstUpdateRef.current) {
+			firstUpdateRef.current = false;
+
+			return;
+		}
+
+		if (validAnalyticsConnection) {
+			chartDispatch({
+				payload: {
+					loading: true,
+				},
+				type: 'SET_PIE_CHART_LOADING',
+			});
+
+			trafficSourcesDataProvider()
+				.then((trafficSources) => {
+					handleDetailPeriodChange(trafficSources, 'paid', true);
```

Comment:
Is it ok to pass always the value "paid"? What if we are in organic detail?

Reply:
I forgot to change it!